### PR TITLE
[Docs]Add karmada-gatekeeper-guide for users

### LINKED ISSUE #1614: [Docs] tutorial of working with Open Policy Agent
**What would you like to be added**:
We need a demo to run OPA on the Karmada control plane to see if there is any change needed. And after that, we can provide a tutorial to describe how to run OPA on Karmada. So the tasks would be:
- [ ] Run a demo with OPA( by precisely [Gatekeeper](https://open-policy-agent.github.io/gatekeeper/website/docs/howto) )
- [ ] Make changes to Karmada if there are any
- [ ] Make a tutorial document

Reference:
- [How to use Gatekeeper](https://open-policy-agent.github.io/gatekeeper/website/docs/howto)
- [Installing Gatekeeper](https://open-policy-agent.github.io/gatekeeper/website/docs/install)
- [Gatekeeper Examples](https://open-policy-agent.github.io/gatekeeper/website/docs/examples)

**Why is this needed**:
The [Open Policy Agent](https://www.openpolicyagent.org/)  is a well-known policy engine and adopted by a large number of users.
Karmada has been asked to support OPA. 

In addition, Karmada still needs to enforce policies for security and compliance purposes.

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add docs for users to run gatekeeper in karmada-controller-plane and use a demo polity. Later, we need to provide some karmada-owned policy to extend karmada ability.

**Which issue(s) this PR fixes**:
Fixes #1614 

**Special notes for your reviewer**:
None

### CODE REVIEW

File: docs/working-with-gatekeeper.md [hunk 1]
```diff
@@ -0,0 +1,2802 @@
...
+```
+hack/local-up-karmada.sh
+```
+
+## Start Gatekeeper
+
+In this case, we will use Gatekeeper v3.7.2. Related deployment files are from [here](https://github.com/open-policy-agent/gatekeeper/blob/release-3.7/deploy/gatekeeper.yaml).
+
+1. Create resource objects of Gatekeeper in karmada controller plane, the content is as follows. 
+
```
Comment (AllenZMC) [inline docs/working-with-gatekeeper.md:19]: If there are no additional modifications to the deployment content, I think just refer to the deployment link. Content can be brief.

Comment (Poor12) [reply docs/working-with-gatekeeper.md:19]: There are something different between karmada-gatekeeper-guide and what from g8sr community. Even though, I still think the draft of the guide will make users confused. Still under discussion.

File: docs/working-with-gatekeeper.md [hunk 2]
```diff
@@ -0,0 +1,2802 @@
...
+      name: gatekeeper-system
+   ---
+   apiVersion: apiextensions.k8s.io/v1
+   kind: CustomResourceDefinition
+   metadata:
+      annotations:
+         controller-gen.kubebuilder.io/version: v0.8.0
+      labels:
+         gatekeeper.sh/system: "yes"
+      name: assign.mutations.gatekeeper.sh
```
Comment (XiShanYongYe-Chang) [inline docs/working-with-gatekeeper.md:41]: This crd file seems so long, can paste the source link address or fold it?

Comment (RainbowMango) [reply docs/working-with-gatekeeper.md:41]: +1

Comment (Poor12) [reply docs/working-with-gatekeeper.md]: Remove crd files and add reference.

**Comment by RainbowMango** [comment]
Propose to organize the content by following menu:
```
# Working with GateKeeper(OPA)    // title
## Prerequisites   // Karmada control plane ready
## GateKeeper Installation
### Install GateKeeper components on host cluster
     <e.g >
```
Deploy namespace: https://github.com/open-policy-agent/gatekeeper/blob/2cf079a74e0a7ca53f6adea3a47024c590fc996b/deploy/gatekeeper.yaml#L1-L8
<descriptions if there any changes needed>
```
### Install GateKeeper APIs on Karmada
## Run demo  
### Create k8srequiredlabels template
### Create k8srequiredlabels constraint
### Create a bad namespace

## Reference
```

**Comment by RainbowMango** [comment]
Propose to organize the content by following menu:
```
# Working with GateKeeper(OPA)    // title
## Prerequisites   // Karmada control plane ready
## GateKeeper Installation
### Install GateKeeper components on host cluster
     <e.g >
```
Deploy namespace: https://github.com/open-policy-agent/gatekeeper/blob/2cf079a74e0a7ca53f6adea3a47024c590fc996b/deploy/gatekeeper.yaml#L1-L8
<descriptions if there any changes needed>
```
### Install GateKeeper APIs on Karmada
## Run demo  
### Create k8srequiredlabels template
### Create k8srequiredlabels constraint
### Create a bad namespace

## Reference
```

File: docs/working-with-gatekeeper.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
+# Working with Gatekeeper(OPA)
+
+[Gatekeeper](https://github.com/open-policy-agent/gatekeeper) , is a customizable admission webhook for Kubernetes that enforces policies executed by the Open Policy Agent (OPA), a policy engine for Cloud Native environments hosted by [Cloud Native Computing Foundation ](https://cncf.io/).
+
+This document gives an example to demonstrate how to use the `Gatekeeper` to manage OPA policy. 
+
+##Start up Karmada clusters
```
Comment (RainbowMango) [inline docs/working-with-gatekeeper.md:7]: Prerequisites

File: docs/working-with-gatekeeper.md (hunk 1 above)
Comment (RainbowMango) [reply docs/working-with-gatekeeper.md:21]: We should add some descriptions about the differences between installing Gatekeeper on `singal` cluster and `Karmada`.

File: docs/working-with-gatekeeper.md [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+             - '*'
+           operations:
+             - CREATE
+             - UPDATE
+           resources:
+             - namespaces
+       sideEffects: None
+       timeoutSeconds: 3
+   ```
+   You need to change the clientconfig from service type to url type for multi-cluster deployment.
```
Comment (RainbowMango) [inline docs/working-with-gatekeeper.md:118]: Can we use `git diff`(or json path, .webhooks[*].clientConfig.url) to highlight the necessary changes for the webhook?

File: docs/working-with-gatekeeper.md [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+   
+   Also, you need to deploy a dummy pod in gatekeeper-system namespace in karmada-apiserver context for Gatekeeper multicluster deployment.
```
Comment (RainbowMango) [inline docs/working-with-gatekeeper.md:120]: Give a description of why we need the dummy pod.

**Review by RainbowMango — COMMENTED** [review]
We should add a more detailed description for each step.

File: docs/working-with-gatekeeper.md (hunk 4 above)
Comment (Poor12) [reply docs/working-with-gatekeeper.md:122]: Add notes for it.

File: docs/working-with-gatekeeper.md (hunk 5 above)
Comment (Poor12) [reply docs/working-with-gatekeeper.md]: Done

File: docs/working-with-gatekeeper.md [hunk 6]
```diff
@@ -0,0 +1,477 @@
+# Working with Gatekeeper(OPA)
+
+[Gatekeeper](https://github.com/open-policy-agent/gatekeeper) , is a customizable admission webhook for Kubernetes that enforces policies executed by the Open Policy Agent (OPA), a policy engine for Cloud Native environments hosted by [Cloud Native Computing Foundation ](https://cncf.io/).
+
+This document gives an example to demonstrate how to use the `Gatekeeper` to manage OPA policy. 
+
+##Prerequisites
+###Start up Karmada clusters
```
Comment (RainbowMango) [inline docs/working-with-gatekeeper.md:8]:
```suggestion
## Prerequisites
### Start up Karmada clusters
```
Markdown requires `white space` after `#` for the title.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**